### PR TITLE
Don't parse globally excluded files

### LINKED ISSUE #1390: Don't parse files in global exclude
What i did was exclude issues from the globally excluded files, but now i realize this is not enough, because you still get the parser error about unsupported objects even if you exclude them. 
You could also use the global exclude to skip parsing autogenerated files.

### PR DESCRIPTION
**Pull request by FreHu**
resolves #1390

### CODE REVIEW

File: packages/core/src/objects/_abap_object.ts [hunk 1]
```diff
@@ -43,6 +46,28 @@ export abstract class ABAPObject extends AbstractObject {
...
+
+    const globalExcludePatterns = (globalExclude).map(pattern => new RegExp(pattern, "i"));
+    const afterExclude: IFile[] = [];
+
+    for (const file of files) {
+      for (const exclude of globalExcludePatterns) {
+        if (exclude.exec(file.getFilename())) {
+          continue;
+        } else {
+          afterExclude.push(file);
```
Comment (larshp) [inline packages/core/src/objects/_abap_object.ts:63]: if there are multiple globalExcludePatterns, then the files will be pushed multiple times?

Comment (FreHu) [reply packages/core/src/objects/_abap_object.ts]: fixed, also cleaned up some of the logic

**Comment by larshp** [comment]
another idea, what if https://github.com/abaplint/abaplint/blob/master/packages/core/src/registry.ts#L153 just did not add the file?

**Comment by FreHu** [comment]
I can do it, whatever works for you. 

I would also not run rules on excluded (meaning rule exclude) files instead of clearing them afterwards, but I'm not sure if this can be handled elegantly in a single place as rules work on objects, not files.

**Comment by larshp** [comment]
Lets try not adding the file, ie skip the file in `addFiles`, I think this will give less code, and less logic in central methods

File: packages/core/src/registry.ts [hunk 2]
```diff
@@ -267,16 +283,35 @@ export class Registry implements IRegistry {
...
   private parsePrivate(input: IObject) {
     if (input instanceof ABAPObject) {
-      const result = input.parse(this.getConfig().getVersion(), this.getConfig().getSyntaxSetttings().globalMacros);
+      const config = this.getConfig();
+      const result = input.parse(config.getVersion(), config.getSyntaxSetttings().globalMacros);
       ParsingPerformance.push(input, result);
     }
   }
 
+  private withoutGloballyExcludedFiles(issues: Issue[], globalExclude: string[]): Issue[] {
```
Comment (larshp) [inline packages/core/src/registry.ts]: then this method can be removed?

Comment (FreHu) [reply packages/core/src/registry.ts]: did it just as you commented

Comment (larshp) [reply packages/core/src/registry.ts]: okay, I were looking at an old version 👍

**Comment by larshp** [comment]
thanks 👍

**Pull request merged**